src/rail/estimation/algos/deepdisc.py

```diff
@@ -72,86 +68,66 @@ def train(config, all_metadata, train_head=True):
     eval_loader = d2data.build_detection_test_loader(
         all_metadata[eval_slice], mapper=mapper, batch_size=batch_size
     )
+    
+    
+    cfg = LazyConfig.load(cfgfile)
+    cfg.OUTPUT_DIR = output_dir
 
-    if train_head:
-        cfg.train.init_checkpoint = None
-
-        model = instantiate(cfg.model)
+    
+    model = return_lazy_model(cfg, train_head)
+    cfg.optimizer.params.model = model
+    optimizer = solver.build_optimizer(cfg, model)
 
-        for param in model.parameters():
-            param.requires_grad = False
-        # Phase 1: Unfreeze only the roi_heads
-        for param in model.roi_heads.parameters():
-            param.requires_grad = True
-        # Phase 2: Unfreeze region proposal generator with reduced lr
-        for param in model.proposal_generator.parameters():
-            param.requires_grad = True
 
-        model.to(cfg.train.device)
-        model = create_ddp_model(model, **cfg.train.ddp)
+    saveHook = return_savehook(run_name)
+    lossHook = return_evallosshook(val_per, model, eval_loader)
+    schedulerHook = return_schedulerhook(optimizer)
+    #hookList = [lossHook, schedulerHook, saveHook]
+    hookList = [schedulerHook, saveHook]
 
-        cfg.optimizer.params.model = model
-        cfg.optimizer.lr = 0.001
-        # optimizer = return_optimizer(cfg)
-        optimizer = solver.build_optimizer(cfg_loader, model)
-        cfg_loader.SOLVER.MAX_ITER = e1  # for DefaultTrainer
+    if train_head:
 
-        saveHook = return_savehook(output_name)
-        lossHook = return_evallosshook(val_per, model, eval_loader)
-        schedulerHook = return_schedulerhook(optimizer)
-        hookList = [lossHook, schedulerHook, saveHook]
+        cfg.SOLVER.MAX_ITER = e1  # for DefaultTrainer
 
         trainer = return_lazy_trainer(
-            model, training_loader, optimizer, cfg, cfg_loader, hookList
+            model, training_loader, optimizer, cfg, hookList
         )
 
         trainer.set_period(print_frequency)
 
-        trainer.train(0, head_iters)
+        trainer.train(0, e1)
 
         if comm.is_main_process():
-            np.save(output_dir + output_name + "_losses", trainer.lossList)
-            # np.save(output_dir + output_name + "_val_losses", trainer.vallossList)
+            np.save(output_dir + run_name + "_losses", trainer.lossList)
+            # np.save(output_dir + run_name + "_val_losses", trainer.vallossList)
```

Comment:
Will this need to be uncommented and conditionally executed when we make `lossHook` conditional?